src/tty.rs

```diff
@@ -179,15 +180,21 @@ fn get_pw_entry(buf: &mut [i8; 1024]) -> Passwd {
 }
 
 /// Create a new tty and return a handle to interact with it.
-pub fn new<T: ToWinsize>(config: &Config, size: T) -> Pty {
+pub fn new<T: ToWinsize>(config: &Config, options: &Options, size: T) -> Pty {
     let win = size.to_winsize();
     let mut buf = [0; 1024];
     let pw = get_pw_entry(&mut buf);
 
     let (master, slave) = openpty(win.ws_row as _, win.ws_col as _);
 
+    // Shell is selected in the order below:
+    //   1. command line argument,
+    //   2. configuration file, or
+    //   3. system user shell
     let default_shell = Shell::new(pw.shell);
-    let shell = config.shell().unwrap_or(&default_shell);
+    let shell = options.shell().unwrap_or_else(|| {
```

Comment:
Can this be cleaned up at all? I'm surprised the nested closure is necessary. I was expecting something like this to work:

```rust
let shell = options.shell()
    .ok_or_else(|| config.shell())
    .unwrap_or(&default_shell);
```